### ISSUE TITLE
gtk::ImageBuilder::icon_size() should take a gtk::IconSize instead of i32

### ISSUE THREAD

**Issue by mmstick** [gtk-rs/gtk#837]

**Comment by sdroege** [comment]
Indeed! Unfortunately the GObject property has the wrong type, so the autogenerated code is also wrong.

This would need some changes in gir to be able to opt-out of builders for specific properties. And then that specific property could be manually implemented. CC @EPashkin @antoyo

**Comment by antoyo** [comment]
Why should it be a `IconSize`?
In GTK+, [it is not](https://developer.gnome.org/gtk3/stable/GtkImage.html#GtkImage--icon-size).

**Comment by sdroege** [comment]
> Why should it be a `IconSize`?
> In GTK+, [it is not](https://developer.gnome.org/gtk3/stable/GtkImage.html#GtkImage--icon-size).

"Symbolic size to use for stock icon, icon set or named icon." says the docs. The symbolic size is exactly that, `GtkIconSize` :) The default value (4) is `GTK_ICON_SIZE_BUTTON`.

**Comment by sdroege** [comment]
And it's a bug in GTK that it uses the wrong type for the property. Creating a MR for GTK4 now so it can at least be fixed there.

**Comment by sdroege** [comment]
> And it's a bug in GTK that it uses the wrong type for the property. Creating a MR for GTK4 now so it can at least be fixed there.

Already fixed there, good

**Comment by antoyo** [comment]
So, is there anyting to fix in gir?
Or would it be fixed automatically with gtk4?

**Comment by sdroege** [comment]
Will be fixed in gtk4, but would be nice to work around for gtk3.
